src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
JS indentation issue;

Suggestion: not required but great opportunity to use .map to store already calculated info and avoid reprocessing:

```js
const children = people.map(
  p => (
    {
      ...p,
      motherObj: (
        (!onlyWithSon || p.sex === 'm')
        && people.find(m => m.name === p.mother)
      )
    }
  )
).filter(el => el.motherObj);

const sum = children.reduce(
  (acc, current) => (
    acc + current.born - current.motherObj.born
  ),
  0
);
```